python/ray/function_manager.py

```diff
@@ -437,6 +446,16 @@ def get_execution_info(self, driver_id, function_descriptor):
             A FunctionExecutionInfo object.
         """
         function_id = function_descriptor.function_id
+        if self._worker.load_code_from_local:
+            # For local code driver id is not necessary.
+            driver_id = ray.DriverID.nil()
+            if len(function_descriptor.class_name) == 0:
+                # This is a normal remote function.
+                self._load_function_from_local(function_descriptor)
+                self._num_task_executions[driver_id][function_id] = 0
+            else:
+                # For actor, it should be loaded before this function.
+                return self._function_execution_info[driver_id][function_id]
```

Comment:
Did you forget to return in the `if` branch?
However, I think maybe we can refactor like the following:
```python
if self._worker.load_code_from_local:
    driver_id = DriverID.nil()
    if not function_descriptor.class_name:  # maybe add a `function_descriptor.is_actor_method()` helper function.
        self._load_function_from_local(function_descriptor)
        self._num_task_executions[driver_id][function_id] = 0
else:
    with profiling.profile("wait_for_function"):
        self._wait_for_function(function_descriptor, driver_id)
try:
    info = self._function_execution_info[driver_id][function_id]
except KeyError as e:
    message = ("Error occurs in get_execution_info: "
               "driver_id: %s, function_descriptor: %s. Message: %s" %
               (driver_id, function_descriptor, e))
    raise KeyError(message)
return info
```

Reply:
Also, `self._num_task_executions[driver_id][function_id] = 0` can be moved to `_load_function_from_local`.